File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Protocols/ProtocolRole.cs
using System;
using VecompSoftware.DocSuiteWeb.Entity.Commons;
using VecompSoftware.DocSuiteWeb.Repository.Entity;

namespace VecompSoftware.DocSuiteWeb.Entity.Protocols
{

    public class ProtocolRole : DSWBaseEntity, IUnauditableEntity
    {
        #region [ Constructor ]
        public ProtocolRole() : this(Guid.NewGuid()) { }

        public ProtocolRole(Guid uniqueId)
            : base(uniqueId)
        {
        }
        #endregion

        #region [ Properties ]
        public short Year { get; set; }
        public int Number { get; set; }
        public string Rights { get; set; }
        public string Note { get; set; }
        public string Type { get; set; }
        public string DistributionType { get; set; }
        public ProtocolRoleNoteType? NoteType { get; set; }
        public ProtocolRoleStatus Status { get; set; }

        #endregion

        #region [ Navigation Properties ]
        public virtual Role Role { get; set; }
        public virtual Protocol Protocol { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Protocol/ProtocolRole.vb
<Serializable()>
Public Class ProtocolRole
    Inherits AuditableDomainObject(Of Guid)

#Region " Fields "
    Private _protocol As Protocol
#End Region

#Region " Properties "

    Public Overridable Property Year As Short

    Public Overridable Property Number As Integer

    Public Overridable Property Rights As String

    Public Overridable Property Note As String

    Public Overridable Property DistributionType As String

    Public Overridable Property Type As String

    Public Overridable Property NoteType As ProtocolRoleNoteType?

    Public Overridable Property Status As ProtocolRoleStatus

    Public Overridable Property Protocol As Protocol
        Get
            Return _protocol
        End Get
        Set(ByVal value As Protocol)
            _protocol = value
            Year = value.Year
            Number = value.Number
        End Set
    End Property

    Public Overridable Property Role As Role

#End Region

#Region " Constructor "

    Public Sub New()
        Status = ProtocolRoleStatus.ToEvaluate
    End Sub

#End Region

End Class


